### PR TITLE
[i18n-ar] Translated file : `docs/source/ar/training.md` into Arabic

### PR DESCRIPTION
**Pull request by AhmedAlmaghz**
## What does this PR do?
Translated the `docs/source/ar/training.md` file of the documentation to `Arabic العربية`. I use transformers for my side-project and wanted to contribute. As a beginner, this seemed appropriate.

### Some questions:

- I can't access the pr documentation endpoint for `Arabic العربية`. Is there a missing step?
- I left all the other files alone after copying the English docs. Is this acceptable for future updates or should I add contents to the `_toctree.yml` only when translation is complete? Should I delete all other files for a cleaner review?
- In `Arabic`, people use both `Transformer` and `المحولات` to describe the model. However, as this is more of a product of HuggingFace I opted not to translate it. May you please let me know which you would prefer?

Thank you in advance for your review.

Part of #32435

### Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/) ?Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

 ### Who can review?
 @stevhliu , @MKhalusova , @abodacs, may you please review this PR?

### CODE REVIEW

File: docs/source/ar/training.md [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+
+<Tip>
+
+سترى تحذيرًا بشأن بعض أوزان النموذج المُدرب مسبقًا التي لم يتم استخدامها وبعض الأوزان التي تم تهيئتها بشكل عشوائي. لا تقلق، هذا أمر طبيعي تمامًا! يتم التخلص من رأس النموذج المُدرب مسبقًا لشبكة BERT، ويتم استبداله برأس تصنيف مُهيأ بشكل عشوائي. سوف تقوم بضبط رأس النموذج الجديد هذا على مهمة تصنيف التسلسل، ونقل معرفة النموذج المُدرب مسبقًا إليه.
+
+</Tip>
+
+### فرط المعالم
+
+بعد ذلك، قم بإنشاء فئة [`TrainingArguments`] والتي تحتوي على جميع فرط المعالم التي يمكنك ضبطها بالإضافة إلى أعلام لتنشيط خيارات التدريب المختلفة. بالنسبة لهذا البرنامج التعليمي، يمكنك البدء بفرط المعالم الافتراضية [hyperparameters](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments)، ولكن لا تتردد في تجربتها للعثور على الإعدادات المثلى.
```
Comment (abodacs) [inline docs/source/ar/training.md:85]:
```suggestion
بعد ذلك، قم بإنشاء كائن من فئة [`TrainingArguments`] والتي تحتوي على جميع العوامل والمتغيرات التي يمكنك ضبطها بالإضافة إلى خيارات تنشيط التدريب المختلفة. بالنسبة لهذا البرنامج التعليمي، يمكنك البدء بمعاملات التدريب الافتراضية [hyperparameters](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments)، ولكن لا تتردد في تجربتها للعثور على الإعدادات المثلى.
```

File: docs/source/ar/training.md [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+
+حدد مكان حفظ نقاط الفحص من تدريبك:
```
Comment (abodacs) [inline docs/source/ar/training.md:87]:
```suggestion
حدد مكان حفظ النسخ من تدريبك:
```

File: docs/source/ar/training.md [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+
+```py
+>>> from transformers import TrainingArguments
+
+>>> training_args = TrainingArguments(output_dir="test_trainer")
+```
+
+### التقييم
+
+لا يقوم [`Trainer`] تلقائيًا بتقييم أداء النموذج أثناء التدريب. ستحتاج إلى تمرير دالة إلى [`Trainer`] لحساب الإبلاغ عن المقاييس. توفر مكتبة [🤗 Evaluate](https://huggingface.co/docs/evaluate/index) دالة [`accuracy`](https://huggingface.co/spaces/evaluate-metric/accuracy) بسيطة يمكنك تحميلها باستخدام الدالة [`evaluate.load`] (راجع هذا [الدليل السريع](https://huggingface.co/docs/evaluate/a_quick_tour) لمزيد من المعلومات):
```
Comment (abodacs) [inline docs/source/ar/training.md:97]:
```suggestion
لا يقوم [`Trainer`] تلقائيًا بتقييم أداء النموذج أثناء التدريب. ستحتاج إلى تمرير دالة إلى [`Trainer`] لحساب وإبلاغ المقاييس. توفر مكتبة [🤗 Evaluate](https://huggingface.co/docs/evaluate/index) دالة [`accuracy`](https://huggingface.co/spaces/evaluate-metric/accuracy) بسيطة يمكنك تحميلها باستخدام الدالة [`evaluate.load`] (راجع هذا [الدليل السريع](https://huggingface.co/docs/evaluate/a_quick_tour) لمزيد من المعلومات):
```

File: docs/source/ar/training.md [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+
+```py
+>>> import numpy as np
+>>> import evaluate
+
+>>> metric = evaluate.load("accuracy")
+```
+
+قم بالاتصال بـ [`~evaluate.compute`] على `metric` لحساب دقة تنبؤاتك. قبل تمرير تنبؤاتك إلى `compute`، تحتاج إلى تحويل logits إلى تنبؤات (تذكر أن جميع نماذج 🤗 Transformers تعيد logits):
```
Comment (abodacs) [inline docs/source/ar/training.md:106]:
```suggestion
استدعِ دالة [`~evaluate.compute`] على `metric` لحساب دقة تنبؤاتك. قبل تمرير تنبؤاتك إلى دالة `compute`، تحتاج إلى تحويل  النتائج الخام logits إلى تنبؤات نهائية (تذكر أن جميع نماذج 🤗 Transformers تعيد نتائج الخام logits):
```

File: docs/source/ar/training.md [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+
+```py
+>>> def compute_metrics(eval_pred):
+...     logits، labels = eval_pred
+...     التوقعات = np.argmax (logits، المحور = -1)
```
Comment (abodacs) [inline docs/source/ar/training.md:111]:
```suggestion
...     predictions = np.argmax(logits, axis=-1)
```

File: docs/source/ar/training.md [hunk 6]
```diff
@@ -0,0 +1,412 @@
...
+...     return metric.compute (التنبؤات = التوقعات، المراجع = العلامات)
```
Comment (abodacs) [inline docs/source/ar/training.md:112]:
```suggestion
...     return metric.compute(predictions=predictions, references=labels)

File: docs/source/ar/training.md [hunk 7]
```diff
@@ -0,0 +1,412 @@
...
+```
+
+إذا كنت ترغب في مراقبة مقاييس التقييم الخاصة بك أثناء الضبط الدقيق، فحدد معلمة `eval_strategy` في حجج التدريب الخاصة بك للإبلاغ عن المقياس في نهاية كل حقبة:
```
Comment (abodacs) [inline docs/source/ar/training.md:115]:
```suggestion
إذا كنت ترغب في مراقبة مقاييس التقييم الخاصة بك أثناء الضبط الدقيق، فحدد معلمة `eval_strategy` في معاملات التدريب الخاصة بك لإظهار مقياس التقييم في نهاية كل حقبة تدريبه:
```

File: docs/source/ar/training.md [hunk 8]
```diff
@@ -0,0 +1,412 @@
...
+
+```py
+>>> from transformers import TrainingArguments, Trainer
+
+>>> training_args = TrainingArguments(output_dir="test_trainer", eval_strategy="epoch")
+```
+
+### المدرب
+
+قم بإنشاء كائن [`Trainer`] باستخدام نموذجك، وحجج التدريب، ومجموعات البيانات التدريبية والاختبارية، ودالة التقييم:
```
Comment (abodacs) [inline docs/source/ar/training.md:125]:
```suggestion
قم بإنشاء كائن [`Trainer`] باستخدام نموذجك، ومعاملات التدريب، ومجموعات البيانات التدريبية والاختبارية، ودالة التقييم:
```

File: docs/source/ar/training.md [hunk 9]
```diff
@@ -0,0 +1,412 @@
...
+نظرًا لأنه مهمة تصنيف نص ثنائي بسيطة، وسنأخذ فقط قسم التدريب الآن.
+
+```py
+from datasets import load_dataset
+
+dataset = load_dataset("glue"، "cola")
+dataset = dataset ["train"] # خذ فقط قسم التدريب الآن
+```
+
+بعد ذلك، قم بتحميل أداة تعلم علامات الترقيم وقم بترميز البيانات كمصفوفات NumPy. لاحظ أن التصنيفات هي بالفعل قائمة من 0 و 1،
```
Comment (abodacs) [inline docs/source/ar/training.md:168]:
```suggestion
بعد ذلك، قم بتحميل أداة المُجزّئ اللغوي وقم بترميز البيانات كمصفوفات NumPy. لاحظ أن التصنيفات هي بالفعل قائمة من 0 و 1،
```

File: docs/source/ar/training.md [hunk 10]
```diff
@@ -0,0 +1,412 @@
...
+
+أنت لست مضطرًا لتمرير دالة خسارة إلى نماذجك عند تجميعها! تختار نماذج Hugging Face تلقائيًا
+دالة خسارة مناسبة لمهمتها وهندسة نموذجها إذا تُركت هذه الحجة فارغة. يمكنك دائمًا
+تجاوز ذلك عن طريق تحديد دالة خسارة بنفسك إذا كنت تريد ذلك!
+
+</Tip>
+
+يعمل هذا النهج بشكل رائع لمجموعات البيانات الصغيرة، ولكن بالنسبة لمجموعات البيانات الأكبر، فقد تجد أنه يصبح مشكلة. لماذا؟
+لأن المصفوفة المرمزة والتصنيفات يجب أن يتم تحميلها بالكامل في الذاكرة، ولأن NumPy لا يتعامل مع
+المصفوفات "المسننة"، لذا يجب أن يتم ترميز كل عينة بالطول الأقصى للعينة في مجموعة البيانات بأكملها. سيؤدي ذلك إلى زيادة حجم المصفوفة لديك، وستبطئ جميع رموز الترقيم من عملية التدريب أيضًا!
```
Comment (abodacs) [inline docs/source/ar/training.md:207]:
```suggestion
المصفوفات"غير المنتظمة"، لذا حشو كل عينة  إلى طول أطول عينة في مجموعة البيانات بأكملها. سيؤدي ذلك إلى زيادة حجم المصفوفة لديك، وستبطئ الرموز الزائده من عملية التدريب أيضًا!
```

File: docs/source/ar/training.md [hunk 11]
```diff
@@ -0,0 +1,412 @@
...
+### تحميل البيانات كـ tf.data.Dataset
+
+إذا كنت تريد تجنب إبطاء التدريب، فيمكنك تحميل بياناتك كـ `tf.data.Dataset` بدلاً من ذلك. على الرغم من أنه يمكنك كتابة خط أنابيب `tf.data` الخاص بك إذا كنت تريد، إلا أن لدينا طريقتين مختصرتين للقيام بذلك:
+- [`~TFPreTrainedModel.prepare_tf_dataset`]: هذه هي الطريقة التي نوصي بها في معظم الحالات. نظرًا لأنه طريقة
+على نموذجك، فيمكنه فحص النموذج لتحديد الأعمدة القابلة للاستخدام كمدخلات للنموذج تلقائيًا،
+واستبعاد الأعمدة الأخرى لإنشاء مجموعة بيانات أبسط وأكثر كفاءة.
+- [`~datasets.Dataset.to_tf_dataset`]: هذه الطريقة أكثر أساسية، وهي مفيدة عندما تريد التحكم بدقة في كيفية
+إنشاء مجموعة البيانات الخاصة بك، عن طريق تحديد أعمدة `columns` و `label_cols` المحددة التي سيتم تضمينها.
+
+قبل أن تتمكن من استخدام [`~TFPreTrainedModel.prepare_tf_dataset`]، ستحتاج إلى إضافة نواتج أداة تعلم علامات الترقيم إلى مجموعة البيانات الخاصة بك كأعمدة، كما هو موضح في
```
Comment (abodacs) [inline docs/source/ar/training.md:218]:
```suggestion
قبل أن تتمكن من استخدام [`~TFPreTrainedModel.prepare_tf_dataset`]، ستحتاج إلى إضافة مخرجات المُجزئ إلى مجموعة البيانات الخاصة بك كأعمدة، كما هو موضح في
```

File: docs/source/ar/training.md [hunk 12]
```diff
@@ -0,0 +1,412 @@
...
+```
+
+تذكر أن مجموعات بيانات Hugging Face يتم تخزينها على القرص بشكل افتراضي، لذا فلن يؤدي ذلك إلى تضخيم استخدام الذاكرة لديك! بمجرد إضافة الأعمدة، يمكنك بث الدفعات من مجموعة البيانات وإضافة الترميز إلى كل دفعة، مما يقلل بشكل كبير من عدد رموز الترقيم مقارنة بترميز مجموعة البيانات بأكملها.
+
+
+```py
+>>> tf_dataset = model.prepare_tf_dataset(dataset["train"], batch_size=16, shuffle=True, tokenizer=tokenizer)
+```
+
+لاحظ أنه في عينة التعليمات البرمجية أعلاه، تحتاج إلى تمرير أداة تعلم علامات الترقيم إلى `prepare_tf_dataset` حتى تتمكن من ترميز الدفعات بشكل صحيح أثناء تحميلها.
```
Comment (abodacs) [inline docs/source/ar/training.md:237]:
```suggestion
لاحظ أنه في عينة التعليمات البرمجية أعلاه، تحتاج إلى تمرير المُجزئ اللغوي إلى `prepare_tf_dataset` حتى تتمكن من حشو الدُفعات بشكل صحيح أثناء تحميلها.
```

File: docs/source/ar/training.md [hunk 13]
```diff
@@ -0,0 +1,412 @@
...
+إذا كانت جميع العينات في مجموعة البيانات الخاصة بك بنفس الطول ولم يكن الترميز ضروريًا، فيمكنك تخطي هذا الحجة.
```
Comment (abodacs) [inline docs/source/ar/training.md:238]:
```suggestion
إذا كانت جميع العينات في مجموعة البيانات الخاصة بك بنفس الطول ولم يكن الترميز ضروريًا، فيمكنك تخطي هذا المعامل.
```

File: docs/source/ar/training.md [hunk 14]
```diff
@@ -0,0 +1,412 @@
...
+إذا كنت بحاجة إلى القيام بشيء أكثر تعقيدًا من مجرد ترميز العينات (على سبيل المثال، إفساد الرموز للنمذجة اللغوية المُقنعة)،
+فيمكنك استخدام حجة `collate_fn` بدلاً من ذلك لتمرير دالة يتم استدعاؤها لتحويل
```
Comment (abodacs) [inline docs/source/ar/training.md:240]:
```suggestion
فيمكنك استخدام معامل `collate_fn` بدلاً من ذلك لتمرير دالة يتم استدعاؤها لتحويل
```

File: docs/source/ar/training.md [hunk 15]
```diff
@@ -0,0 +1,412 @@
...
+<frameworkcontent>
+<pt>
+<Youtube id="Dh9CL8fyG80"/>
+
+[`Trainer`] يهتم بحلقة التدريب ويسمح لك بضبط نموذج في سطر واحد من التعليمات البرمجية. بالنسبة للمستخدمين الذين يفضلون كتابة حلقة التدريب الخاصة بهم، يمكنك أيضًا ضبط نموذج 🤗 Transformers في PyTorch الأصلي.
+
+في هذه المرحلة، قد تحتاج إلى إعادة تشغيل دفتر الملاحظات الخاص بك أو تنفيذ التعليمات البرمجية التالية لتحرير بعض الذاكرة:
+
+```py
+حذف النموذج
```
Comment (abodacs) [inline docs/source/ar/training.md:267]:
```suggestion
del model
```

File: docs/source/ar/training.md [hunk 16]
```diff
@@ -0,0 +1,412 @@
...
+حذف المدرب
```
Comment (abodacs) [inline docs/source/ar/training.md:268]:
```suggestion
del trainer
```

**Review by abodacs — CHANGES_REQUESTED** [review]
السﻻم عليكم
شكرا لمجهودك
@AhmedAlmaghz 

تمت المراجعة

**Comment by AhmedAlmaghz** [comment]
Thank you @abodacs
